src/vision.cs

```diff
@@ -0,0 +1,792 @@
+
+		[Export ("initWithTargetedImageData:")]
+		IntPtr Constructor (NSData imageData);
+
+		[Export ("initWithTargetedImageData:completionHandler:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSData imageData, [NullAllowed] VNRequestCompletionHandler completionHandler);
+	}
+
+	[TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (VNImageRegistrationRequest))]
+	interface VNHomographicImageRegistrationRequest {
+
+		// Inlined from parent class
+		[Export ("initWithTargetedCVPixelBuffer:")]
+		IntPtr Constructor (CVPixelBuffer pixelBuffer);
+
+		[Export ("initWithTargetedCVPixelBuffer:completionHandler:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (CVPixelBuffer pixelBuffer, [NullAllowed] VNRequestCompletionHandler completionHandler);
+
+		[Export ("initWithTargetedCGImage:")]
+		IntPtr Constructor (CGImage image);
+
+		[Export ("initWithTargetedCGImage:completionHandler:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (CGImage image, [NullAllowed] VNRequestCompletionHandler completionHandler);
+
+		[Export ("initWithTargetedCIImage:")]
+		IntPtr Constructor (CIImage image);
+
+		[Export ("initWithTargetedCIImage:completionHandler:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (CIImage image, [NullAllowed] VNRequestCompletionHandler completionHandler);
+
+		[Export ("initWithTargetedImageURL:")]
+		IntPtr Constructor (NSUrl imageUrl);
+
+		[Export ("initWithTargetedImageURL:completionHandler:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSUrl imageUrl, [NullAllowed] VNRequestCompletionHandler completionHandler);
+
+		[Export ("initWithTargetedImageData:")]
+		IntPtr Constructor (NSData imageData);
+
+		[Export ("initWithTargetedImageData:completionHandler:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSData imageData, [NullAllowed] VNRequestCompletionHandler completionHandler);
+	}
+
+	[TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[Abstract]
+	[BaseType (typeof (NSObject))]
+	interface VNObservation : NSCopying, NSSecureCoding {
+
+		[Export ("uuid", ArgumentSemantic.Strong)]
+		NSUuid Uuid { get; }
+
+		[Export ("confidence")]
+		float Confidence { get; }
+	}
+
+	[TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (VNObservation))]
+	interface VNDetectedObjectObservation {
+
+		[Static]
+		[Export ("observationWithBoundingBox:")]
+		VNDetectedObjectObservation FromBoundingBox (CGRect boundingBox);
+
+		[Export ("boundingBox", ArgumentSemantic.Assign)]
+		CGRect BoundingBox { get; }
+	}
+
+	[TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (VNDetectedObjectObservation))]
+	interface VNFaceObservation {
+
+		[NullAllowed, Export ("landmarks", ArgumentSemantic.Strong)]
+		VNFaceLandmarks2D Landmarks { get; }
+	}
+
+	[TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (VNObservation))]
+	interface VNClassificationObservation {
```

Comment:
from a quick doc read it seems it's something given to you
not something user code should create